Authenticators/Elkarte/elkarteauth.py

```diff
@@ -0,0 +1,887 @@
+            """
+
+            FALL_THROUGH = ""
+            
+            debug('idToTexture for %d', id)
+            if id < cfg.user.id_offset or not cfg.user.avatar_enable:
+                debug('idToTexture %d -> fall through', id)
+                return FALL_THROUGH
+            
+            # Otherwise get the users texture from elkarte
+            bbid = id - cfg.user.id_offset
+            try:
+                sql = 'SELECT avatar FROM %smembers WHERE id_member = %%s' % cfg.database.prefix
+                cur = threadDB.execute(sql, [bbid])
+            except threadDbException:
+                return FALL_THROUGH
+            res = cur.fetchone()
+            cur.close()
+            if not res:
+                debug('idToTexture %d -> user unknown, fall through', id)
+                return FALL_THROUGH
+            avatar = res[0]
+            
+            if not avatar:
+                # Either the user has none or it is in the attachments, check there
+                try:
+                    sql = '''SELECT id_attach, file_hash, filename, attachment_type FROM %sattachments WHERE approved = true AND
+                        (attachment_type = 0 OR attachment_type = 1) AND id_member = %%s''' % cfg.database.prefix
+                    cur = threadDB.execute(sql, [bbid])
+                except threadDbException:
+                    return FALL_THROUGH
+                
+                res = cur.fetchone()
+                cur.close()
+                if not res:
+                    # No uploaded avatar found, seems like the user didn't set one
+                    debug('idToTexture %d -> no texture available for this user, fall through', id)
+                    return FALL_THROUGH
+                
+                fid, fhash, filename, fattachtype = res
+                if cfg.forum.path.startswith('file://'):
+                    # We are supposed to load this from the local fs
+                    avatar_file = cfg.forum.path + 'attachments/%d_%s' % (fid, fhash)
+                elif fattachtype == 0: 
+                    avatar_file = cfg.forum.path + 'index.php?action=dlattach;attach=%d;type=avatar' % fid
+                elif fattachtype == 1:
+                    avatar_file = cfg.forum.path + 'avatars/' + filename 
+            elif "://" in avatar:
+                # ...or it is a external link
+                avatar_file = avatar
+            else:
+                warning("avatar with an unexpected value, fall through")
+                return FALL_THROUGH
+                
+            if avatar_file in self.texture_cache:
+                return self.texture_cache[avatar_file]
+            
+            try:
+                handle = urllib2.urlopen(urllib2.Request(avatar_file, headers={'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (Windows; U; Windows NT 5.1; en-US; rv:1.9.0.7) Gecko/2009021910 Firefox/3.0.7'}))
```

Comment:
I agree about spoofing.
We should not spoof at all. If cloudflare specifically checks user agent and changes behaviour, we should primarily look at cloudflare. For us, we should behave nicely.
If this is something along the lines of "we only host (in specific plans) for webbrowsers" and we make users spoof here, they may get blocked entirely.

I would simply put a notice into the header/readme/faq about the cloudflare issue.